tket/tests/test_TwoQubitCanonical.cpp

```diff
@@ -43,6 +44,33 @@ static void check_get_information_content(const Eigen::Matrix4cd &U) {
   REQUIRE(res.isApprox(U));
 }
 
+SCENARIO("Testing get_matrix_from_2qb_circ") {
+  Circuit c(2);
+  GIVEN("A CX") { c.add_op<unsigned>(OpType::CX, {0, 1}); }
+  GIVEN("A reverse CX") { c.add_op<unsigned>(OpType::CX, {1, 0}); }
+  GIVEN("A Swap") { c.add_op<unsigned>(OpType::SWAP, {0, 1}); }
+  GIVEN("A TK1") { c.add_op<unsigned>(OpType::TK1, {0.3, .2, -.6}, {0}); }
+  GIVEN("A TK2") { c.add_op<unsigned>(OpType::TK2, {0.3, .2, -.6}, {0, 1}); }
+  GIVEN("A reverse TK2") {
+    c.add_op<unsigned>(OpType::TK2, {0.3, .2, -.6}, {1, 0});
+  }
+  GIVEN("A bunch of gates") {
+    c.add_op<unsigned>(OpType::TK1, {0.3, .2, -.6}, {0});
+    c.add_op<unsigned>(OpType::TK1, {0.3, 2.39, 1.6}, {1});
+    c.add_op<unsigned>(OpType::CX, {0, 1});
+    c.add_op<unsigned>(OpType::Vdg, {0});
+    c.add_op<unsigned>(OpType::H, {1});
+    c.add_op<unsigned>(OpType::Tdg, {1});
+    c.add_op<unsigned>(OpType::CX, {1, 0});
+  }
+
+  THEN("The unitaries from tket_sim and get_matrix are identical") {
+    auto u1 = tket_sim::get_unitary(c);
+    auto u2 = get_matrix_from_2qb_circ(c);
+    REQUIRE(u1.isApprox(u2));
```

Comment:
You're not testing that the unitaries are "identical", only that they are "approximately equal".

Which do you actually mean? If one of the functions `tket_sim::get_unitary` or `get_matrix_from_2qb_circ` is a wrapper around the other, then "identical" would be worth testing (_i.e., exact equality of doubles, normally very naughty but justified in rare cases_); but if they're calculated using different but algebraically equivalent formulae then approximations as here are the correct thing to test, since there could always be roundoff.

Reply:
No, they are different methods to calculate the same. I considered replacing `get_matrix_from_2qb_circ` with `tket_sim`, but it would create a dependency on `Simulation` for basically every tket module. And probably some circular dependencies as a consequence too.